src/main/java/com/ericsson/ei/subscription/InformSubscriber.java

```diff
@@ -123,168 +131,223 @@ public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson)
-    private HttpHeaders prepareHeaders(String notificationMeta, JsonNode subscriptionJson) {
+    private HttpHeaders prepareHeaders(String notificationMeta, JsonNode subscriptionJson)
+            throws AuthenticationException {
         HttpHeaders headers = new HttpHeaders();
-        String headerContentMediaType = getSubscriptionField("restPostBodyMediaType", subscriptionJson);
+        String headerContentMediaType = getSubscriptionField("restPostBodyMediaType",
+                subscriptionJson);
         headers.setContentType(MediaType.valueOf(headerContentMediaType));
-        LOGGER.debug("Successfully added header: 'restPostBodyMediaType':'{}'", headerContentMediaType);
+        LOGGER.debug("Successfully added header: 'restPostBodyMediaType':'{}'",
+                headerContentMediaType);
+
         headers = addAuthenticationData(headers, notificationMeta, subscriptionJson);
 
         return headers;
     }
 
     /**
-     * This method adds the authentication details to the headers.
+     * Adds the authentication details to the headers.
      *
      * @param headers
-     * @param notificationMeta
-     *     A String containing a URL
-     * @param subscriptionJson
-     *     Used to extract the authentication type from
+     * @param notificationMeta A String containing a URL
+     * @param subscriptionJson Used to extract the authentication type from
      * @return headers
+     * @throws AuthenticationException
      */
-    private HttpHeaders addAuthenticationData(HttpHeaders headers, String notificationMeta, JsonNode subscriptionJson) {
+    private HttpHeaders addAuthenticationData(HttpHeaders headers, String notificationMeta,
+            JsonNode subscriptionJson) throws AuthenticationException {
         String authType = getSubscriptionField("authenticationType", subscriptionJson);
-        if (!authType.equals("BASIC_AUTH")) {
-            return headers;
-        }
-
         String username = getSubscriptionField("userName", subscriptionJson);
         String password = getSubscriptionField("password", subscriptionJson);
 
-        if (!username.equals("") && !password.equals("")) {
-            String encoding = Base64.getEncoder().encodeToString((username + ":" + password).getBytes());
-            headers.add("Authorization", "Basic " + encoding);
-            LOGGER.debug("Successfully added header for 'Authorization'");
+        boolean authorizationDetailsProvided = isAuthenticationDetailsProvided(authType, username,
+                password);
+        if (!authorizationDetailsProvided) {
+            return headers;
+        }
 
-            // Adding jenkins crumb if any
-            headers = addJenkinsCrumbData(headers, encoding, notificationMeta);
+        String encoding = Base64.getEncoder().encodeToString((username + ":" + password)
+                .getBytes());
+        headers.add("Authorization", "Basic " + encoding);
+        LOGGER.debug("Successfully added header for 'Authorization'");
```

Comment:
Same comment regarding how we log authentication/authorization...  Isn't it more confusing if both are used interchangeably? Maybe stick to one both in documentation/logging and code unless we actually mean something different and want to emphasize this

Reply:
No change due to header being named authorization